cmd/wfx/cmd/root/server_collection.go

```diff
@@ -0,0 +1,389 @@
+		Middlewares: combinedMWs,
+	})
+	server, err := server.NewHTTPServer(cfg, handler)
+	return server, fault.Wrap(err)
+}
+
+func createPluginMiddlewares(pluginDir string) ([]*plugin.Middleware, error) {
+	plugins, err := loadPlugins(pluginDir)
+	if err != nil {
+		return nil, fault.Wrap(err)
+	}
+
+	pluginMWs := make([]*plugin.Middleware, 0, len(plugins))
+	for _, p := range plugins {
+		chErr := make(chan error, 1)
+		mw, err := plugin.NewMiddleware(p, chErr)
+		if err != nil {
+			return nil, fault.Wrap(err)
+		}
+		pluginMWs = append(pluginMWs, mw)
+	}
+	return pluginMWs, nil
+}
+
+type ListenerSettings struct {
+	Host    string
+	Port    int
+	TLSHost string
+	TLSPort int
+	UDSPath string
+}
+
+func createListener(scheme config.Scheme, settings ListenerSettings) (net.Listener, error) {
+	var network, addr string
+	switch scheme {
+	case config.SchemeUnix:
+		network = "unix"
+		addr = settings.UDSPath
+	case config.SchemeHTTP:
+		network = "tcp"
+		addr = fmt.Sprintf("%s:%d", settings.Host, settings.Port)
+	case config.SchemeHTTPS:
+		network = "tcp"
+		addr = fmt.Sprintf("%s:%d", settings.TLSHost, settings.TLSPort)
+	default:
+		return nil, fmt.Errorf("unsupported scheme: %s", scheme)
+	}
+	contextLogger := log.With().Str("network", network).Str("addr", addr).Str("scheme", scheme.String()).Logger()
+	for attempt := 0; attempt < 30; attempt++ {
+		ln, err := net.Listen(network, addr)
+		if err == nil {
+			contextLogger.Debug().Msg("Created new listener")
+			return ln, nil
+		}
+		contextLogger.Err(err).Msg("Failed to create listener")
+		time.Sleep(time.Second)
+	}
+	return nil, errors.New("failed to create listener")
+}
+
+func createMux(cfg *config.AppConfig, server genApi.ServerInterface) *http.ServeMux {
+	swag, _ := genApi.GetSwagger()
+
+	mux := http.NewServeMux()
+	mux.HandleFunc("GET /version", func(w http.ResponseWriter, r *http.Request) { server.GetVersion(w, r) })
+	mux.HandleFunc("GET /health", func(w http.ResponseWriter, r *http.Request) { server.GetHealth(w, r) })
+	mux.HandleFunc("GET /", func(w http.ResponseWriter, _ *http.Request) {
+		w.WriteHeader(http.StatusNotFound)
+		_, _ = w.Write([]byte("The requested resource could not be found.\n\nHint: Check /openapi.json to see available endpoints.\n"))
+	})
+	mux.HandleFunc("GET /openapi.json", func(w http.ResponseWriter, _ *http.Request) {
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusOK)
+		_ = json.NewEncoder(w).Encode(swag)
+	})
+	mux.HandleFunc("GET /download/", func(w http.ResponseWriter, r *http.Request) {
+		rootDir := cfg.SimpleFileserver()
+		enabled := rootDir != ""
+		log.Debug().Bool("enabled", enabled).Msg("Received download request")
+		if enabled {
+			http.StripPrefix("/download", http.FileServer(http.Dir(rootDir))).ServeHTTP(w, r)
+		} else {
+			w.WriteHeader(http.StatusNotFound)
+			w.Header().Set("Content-Type", "application/json")
+			_, _ = w.Write([]byte(`{"code":404,"message":"path /download was not found"}`))
```

Comment:
Hm, "is not served as simple file server is not enabled" or something along those lines would be more telling?